packages/devtools_app/test/performance/frame_analysis/frame_hints_test.dart

```diff
@@ -0,0 +1,326 @@
+import 'package:devtools_app/src/screens/performance/panes/controls/enhance_tracing/enhance_tracing_model.dart';
+import 'package:devtools_app/src/screens/performance/panes/frame_analysis/frame_analysis_model.dart';
+import 'package:devtools_app/src/screens/performance/panes/frame_analysis/frame_hints.dart';
+import 'package:devtools_app/src/screens/performance/performance_controller.dart';
+import 'package:devtools_app/src/screens/performance/performance_model.dart';
+import 'package:devtools_app/src/service/service_manager.dart';
+import 'package:devtools_app/src/shared/globals.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../test_data/performance.dart';
+
+void main() {
+  const windowSize = Size(4000.0, 1000.0);
+
+  group('FrameHints', () {
+    late MockEnhanceTracingController mockEnhanceTracingController;
+    late MockFrameAnalysis mockFrameAnalysis;
+    late MockFramePhase mockBuildPhase;
+    late MockFramePhase mockLayoutPhase;
+    late MockFramePhase mockPaintPhase;
+    late MockFramePhase mockRasterPhase;
+
+    setUp(() {
+      mockEnhanceTracingController = MockEnhanceTracingController();
+      mockFrameAnalysis = MockFrameAnalysis();
+      mockBuildPhase = MockFramePhase();
+      when(mockBuildPhase.title).thenReturn(FramePhaseType.build.eventName);
+      when(mockBuildPhase.type).thenReturn(FramePhaseType.build);
+      mockLayoutPhase = MockFramePhase();
+      when(mockLayoutPhase.title).thenReturn(FramePhaseType.layout.eventName);
+      when(mockLayoutPhase.type).thenReturn(FramePhaseType.layout);
+      mockPaintPhase = MockFramePhase();
+      when(mockPaintPhase.title).thenReturn(FramePhaseType.paint.eventName);
+      when(mockPaintPhase.type).thenReturn(FramePhaseType.paint);
+      mockRasterPhase = MockFramePhase();
+      when(mockRasterPhase.title).thenReturn(FramePhaseType.raster.eventName);
+      when(mockRasterPhase.type).thenReturn(FramePhaseType.raster);
+
+      setGlobal(IdeTheme, IdeTheme());
+      setGlobal(OfflineModeController, OfflineModeController());
+      final fakeServiceManager = FakeServiceManager();
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+      mockConnectedApp(
+        fakeServiceManager.connectedApp!,
+        isFlutterApp: true,
+        isProfileBuild: true,
+        isWebApp: false,
+      );
+    });
+
+    Future<void> pumpHints(
+      WidgetTester tester,
+      FrameAnalysis frameAnalysis,
+    ) async {
+      await tester.pumpWidget(
+        wrapWithControllers(
+          FrameHints(
+            frameAnalysis: frameAnalysis,
+            enhanceTracingController: mockEnhanceTracingController,
+          ),
+          performance: PerformanceController(),
+        ),
+      );
+      expect(find.byType(FrameHints), findsOneWidget);
+    }
+
+    void _mockFrameAnalysis({
```

Comment:
'_' in the beginning does not have effect here, because it is in method body

Reply:
it doesn't do anything for privacy, but it does avoid a naming collision with the mock object `mockFrameAnalysis`